Yoast/wordpress-seo
# Introduce collapsable paper blocks in search appearance

## Summary

This PR can be summarized in the following changelog entry:

* Introduces collapsable sections in the search appearance tabs.

## Relevant technical choices:

* Added a presenter class that can create collapsable blocks. this is already in use for the following tabs:

    - taxonomies: the current implementation is broken and commented out since the view still contain logic that needs to be converted.
    - Archives: the presenter class is fully implemented but misses help panels. Look into the state of the collapsable as well, since they are currently all expanded by default at page reload.
    - Rss: Works perfectly according to the issue description but misses a help panel in the title. Maybe the 2 headings will have to be separated.
    - Breadcrumbs: the presenter class is fully implemented but the view still contains too much logic In my opinion.
    - General: The presenter class is not implemented, but the styling is already corresponding to the issue. Maybe the headings will have to be separated.
    - Post Types: no work is done on this one, but it had already collapsable paper blocks so it only needs a refactor to use the presenter class,
    - Media:  The presenter class is implemented and works but it still misses the help icons at the title.

Overall it just is a lot of code moved in view files that will be wrapped by a the paper-collapsable if the presenter class is implemented.

## Test instructions

This PR can be tested by following these steps:

* Verify that everything (except category urls under taxonomies) is wrapped in the paper styling.
* Make sure the recommended replacement variables are still working (because of trunk merge).


## Quality assurance


Fixes #10110
